decoder/api_gym.go

```diff
@@ -0,0 +1,144 @@
+	var args []any
+
+	// Always include enabled = 1
+	whereConditions = append(whereConditions, "enabled = 1")
+
+	for _, filter := range search.Filters {
+		// Name filter
+		if filter.Name != nil && strings.TrimSpace(*filter.Name) != "" {
+			whereConditions = append(whereConditions, "name LIKE ?")
+			args = append(args, "%"+escapeLike(strings.TrimSpace(*filter.Name))+"%")
+		}
+
+		// Description filter
+		if filter.Description != nil && strings.TrimSpace(*filter.Description) != "" {
+			whereConditions = append(whereConditions, "description LIKE ?")
+			args = append(args, "%"+escapeLike(strings.TrimSpace(*filter.Description))+"%")
+		}
+
+		// Location distance filter
+		if filter.LocationDistance != nil {
+			locDist := *filter.LocationDistance
+			lat, lon := locDist.Location.Latitude, locDist.Location.Longitude
+			distance := locDist.Distance
+
+			// Calculate bounding box for performance
+			latDelta := distance / 111_045.0
+			lonScale := math.Cos(lat * math.Pi / 180)
+			if lonScale < 1e-6 {
+				lonScale = 1e-6
+			}
+			lonDelta := distance / (111_045.0 * lonScale)
+
+			latMin, latMax := lat-latDelta, lat+latDelta
+			lonMinRaw, lonMaxRaw := lon-lonDelta, lon+lonDelta
+
+			lonMin := geo.NormalizeLon(lonMinRaw)
+			lonMax := geo.NormalizeLon(lonMaxRaw)
+			crossesAM := lonMin > lonMax
+
+			// Add bounding box conditions
+			whereConditions = append(whereConditions, "lat BETWEEN ? AND ?")
+			args = append(args, latMin, latMax)
+
+			if crossesAM {
+				whereConditions = append(whereConditions, "(lon >= ? OR lon <= ?)")
+				args = append(args, lonMin, lonMax)
+			} else {
+				whereConditions = append(whereConditions, "lon BETWEEN ? AND ?")
+				args = append(args, lonMin, lonMax)
+			}
+
+			// Add precise distance condition
+			whereConditions = append(whereConditions, "ST_Distance_Sphere(POINT(lon, lat), POINT(?, ?)) <= ?")
+			args = append(args, lon, lat, distance)
+		}
+
+		// Bounding box filter
+		if filter.Bbox != nil {
+			bbox := *filter.Bbox
+			latMin := math.Min(bbox.MinLat, bbox.MaxLat)
+			latMax := math.Max(bbox.MinLat, bbox.MaxLat)
+
+			lonMin := geo.NormalizeLon(bbox.MinLon)
+			lonMax := geo.NormalizeLon(bbox.MaxLon)
+			crossesAM := lonMin > lonMax
+
+			whereConditions = append(whereConditions, "lat BETWEEN ? AND ?")
+			args = append(args, latMin, latMax)
+
+			if crossesAM {
+				whereConditions = append(whereConditions, "(lon >= ? OR lon <= ?)")
+				args = append(args, lonMin, lonMax)
+			} else {
+				whereConditions = append(whereConditions, "lon BETWEEN ? AND ?")
+				args = append(args, lonMin, lonMax)
+			}
+		}
+	}
+
+	// Build the final query
+	whereClause := strings.Join(whereConditions, " AND ")
+	rawSQL := fmt.Sprintf(`
+		SELECT id
+		FROM gym
+		WHERE %s
+		ORDER BY id ASC
+		LIMIT ?
+	`, whereClause)
+
+	args = append(args, search.Limit)
+	q := dbDetails.GeneralDb.Rebind(rawSQL)
+
+	var ids []string
+	if err := dbDetails.GeneralDb.SelectContext(ctx, &ids, q, args...); err != nil {
```

Comment:
I assume that query got that heavy that it doesn't make sense to use the in-memory - cache ?! :)

or wait: you do a SQL query to select gym ids, afterwards you use it to access the in-memory state?